Iterate the app

alexmuller opened the pull request:
## Add a healtcheck path

Which tests that MongoDB is running and available.
## Don't save the original-policy from the report

The `original-policy` field contains the entire `Content-Security-Policy` header, which will in most cases be very large (so it will fail validation) and mostly not useful.

alexmuller committed:
Add a healthcheck path

Which tests that MongoDB is running and available.

alexmuller committed:
Change default port to production port

alexmuller committed:
Don't save the original-policy from the report

The original-policy field contains the entire Content-Security-Policy
header, which will in most cases be very large (so it will fail
validation) and mostly not useful.

alexmuller committed:
Change environment variable for MongoDB

This is a comma-separated list of MongoDB node names, not a URL.

KushalP commented on handlers.go:
```diff
@@ -102,3 +101,24 @@ func ReportHandler(w http.ResponseWriter, req *http.Request) {
+		return
+	}
+
+	w.Write([]byte("OK"))
```
It would be good to have something a little more explicit should one of us (people) hit that and want to see the explicit endpoints. For the crawler we provide the specific endpoints as JSON: https://github.com/alphagov/govuk_crawler_worker/blob/507009d7a2eb5e5855633d6864743586cdc4db5b/health_check.go#L12-15

The equivalent here would look something like the following:

``` json
{"mongo": true}
```

KushalP commented:
This looks good, but it would great if you can add a test to make sure the health check does what it says on the tin. A good example of this can found in the `publishing-api` which has a similar health check and a test for it that uses `httptest.NewServer` to wrap the handler: https://github.com/alphagov/publishing-api/blob/6b89bb82add568d8c4ffcc5470f0fbd68d561501/healthcheck_test.go

You could also use the `httptest.NewRecorder()` that you already use for the other handler. Entirely up to you really, but it's worth nothing that one tests the go http.Handler (NewRecorder) whereas the other will test the route endpoint associated with the mux (NewServer).

alexmuller commented:
@KushalP: Thanks for pointing to the crawler worker, that was really helpful. I've changed the endpoint to return JSON in 266f0429c770a76b417a0b9da9682bc4b7a208ae, but I'm having trouble writing a test for it because whenever the Mongo session can't be established the application panics, so there's no response.

alexmuller committed:
Update Makefile

Symlinking the current directory into _vendor allows us to use
Ginkgo to run tests.

alexmuller committed:
Rewrite tests using Ginkgo

Ginkgo and Gomega give a nice easy way to make assertions

alexmuller committed:
Uppercase the Content Security Policy initialism

The [Go CodeReviewComments doc on initialiasms][1] says that they
should be uppercased, eg that URL should appear as URL not Url.

[1]: https://github.com/golang/go/wiki/CodeReviewComments#initialisms

alexmuller commented:
With help from @KushalP yesterday (thank you!) I've made some changes to this pull request. The commits are in a bit of a weird order in the web interface but they make sense when ordered in the tree.

The major commits (in correct order) are:
- Update Makefile (3d7615fced54963f2f80490c0a2d3062dd998bfc)
- Rewrite tests using Ginkgo (1a50658fcb72e318cd0b0fe0dbfc1287b776cf38)
- Add a tested healthcheck endpoint (0360fda415ebb6ad4f8710cd8366ff309ec43265)

KushalP commented:
I don't know why the tests haven't run and been marked here as passing. But I've just run this locally and everything looks good :+1:
